go/libraries/doltcore/doltdb/gcctx/gc_safepoint_controller.go

```diff
@@ -152,6 +154,10 @@ func (c *GCSafepointController) Waiter(ctx context.Context, thisSession GCRootsP
 		if sess == thisSession {
 			continue
 		}
+		if state.cancel != nil {
+			state.cancel()
+			continue
```

Comment:
This doesn't seem safe. Processing the cancelation is async but the waiter needs to wait until the session command is actually gone. It seems like even after we cancel the context, we need to add the outstanding command to the the waiters list and await it like normal.